### PR TITLE
[metrics]:add metrics for snapshot(kv-count & size)

### PR DESCRIPTION
**Pull request by AndreMouche**
@BusyJay @disksing PTAL

### CODE REVIEW

File: src/raftstore/store/worker/metrics.rs [hunk 1]
```diff
@@ -74,4 +74,15 @@ lazy_static! {
             "tikv_raftstore_hash_duration_seconds",
             "Bucketed histogram of raftstore hash compution duration"
         ).unwrap();
+
+    pub static ref SNAPSHOT_KV_COUNET_GAUGE: Gauge =
```
Comment (BusyJay) [inline src/raftstore/store/worker/metrics.rs]: Should use Histogram instead. Ref: https://prometheus.io/docs/concepts/metric_types/

Comment (disksing) [inline src/raftstore/store/worker/metrics.rs]: COUNET -> COUNT

Comment (AndreMouche) [inline src/raftstore/store/worker/metrics.rs]: Use  Histogram instead & fix spelling mistake @BusyJay @disksing

File: src/raftstore/store/worker/metrics.rs [hunk 2]
```diff
@@ -74,4 +74,19 @@ lazy_static! {
             "tikv_raftstore_hash_duration_seconds",
             "Bucketed histogram of raftstore hash compution duration"
         ).unwrap();
+
+    pub static ref SNAPSHOT_KV_COUNT_HISTOGRAM: Histogram =
+        register_histogram!(
+            "tikv_snapshot_kv_count",
+            "Total number of kv in snapshot",
+             vec![1000.0, 10000.0, 100000.0, 1000000.0, 10000000.0]
```
Comment (siddontang) [inline src/raftstore/store/worker/metrics.rs]: 100, 500, 1000, 5000, 10000, 50000, 100000, 500000, 1000000

File: src/raftstore/store/worker/metrics.rs [hunk 3]
```diff
@@ -74,4 +74,19 @@ lazy_static! {
...
+
+        ).unwrap();
+    pub static ref SNAPSHOT_SIZE_HISTOGRAM: Histogram =
+        register_histogram!(
+            "tikv_snapshot_size",
+            "Size of snapshot",
+             vec![1000.0, 10000.0, 100000.0, 1000000.0, 10000000.0,
```
Comment (siddontang) [inline src/raftstore/store/worker/metrics.rs]: use `exponential_buckets(1024, 2.0, 22).unwrap()`, so we can cover from 1KB to 4GB.

File: src/raftstore/store/worker/metrics.rs (hunk 2 above)
Comment (siddontang) [inline src/raftstore/store/worker/metrics.rs]: `exponential_buckets(100, 2.0, 20).unwrap()`

**Comment by siddontang** [comment]
LGTM

File: src/raftstore/store/worker/metrics.rs [hunk 4]
```diff
@@ -74,4 +74,17 @@ lazy_static! {
             "tikv_raftstore_hash_duration_seconds",
             "Bucketed histogram of raftstore hash compution duration"
         ).unwrap();
+
+    pub static ref SNAPSHOT_KV_COUNT_HISTOGRAM: Histogram =
+        register_histogram!(
+            "tikv_snapshot_kv_count",
+            "Total number of kv in snapshot",
+             exponential_buckets(100.0, 2.0, 20).unwrap() //100,100*2^1,...100M
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/worker/metrics.rs]: The region max size may increase to 1GB someday.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**